book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -367,23 +373,35 @@ handled by `#[derive(Default)]`.
     all its blocks have been `pop`ped
 
 
-### `ChainSet` Type
-[chainset-type]: #chainset-type
+### `MemoryState` Type
+[memorystate-type]: #memorystate-type
 
-The `ChainSet` type represents the set of all non-finalized state. It
+The `MemoryState` type represents the set of all non-finalized state. It
 consists of a set of non-finalized but verified chains and a set of
 unverified blocks which are waiting for the full context needed to verify
 them to become available.
 
-`ChainSet` is defined by the following structure and API:
+`MemoryState` is defined by the following structure and API:
 
 ```rust
-struct ChainSet {
-    chains: BTreeSet<Chain>,
+/// The state of the chains in memory, incuding queued blocks.
+#[derive(Debug, Default)]
+pub struct MemoryState {
```

Comment:
@teor2345 I applied your change to split up the queued blocks here but I went ahead and switch it and the `SledState` to different names. lmk what you think